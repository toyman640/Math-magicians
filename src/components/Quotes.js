import React, { useState, useEffect } from 'react';

function Quotes() {
  const [quote, setQuote] = useState(null);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(null);
  useEffect(() => {
    const GetData = async () => {
      try {
        const result = await fetch('https://api.api-ninjas.com/v1/quotes?limit=1&category=hope', { method: 'GET', headers: { 'X-Api-Key': 'hFl61f9hCHCBsGXi0S10VA==2C9iKOYx8LbDvxbP' } });
        if (!result.ok) {
          throw new Error('Failed to fetch data');
        }
        const ParsedQuote = await result.json();
        setQuote(ParsedQuote);
        setIsLoading(false);
      } catch (error) {
        setError(error.message);
      } finally {
        setIsLoading(false);
      }
    };
    GetData();
  }, []);
  if (isLoading) {
    return <p className="Quote">Loading...</p>;
  }

  if (error) {
    return (
      <p className="Quote">
        Error:
        {error}
      </p>
    );
  }
  return (
    <div className="Quote">
      { quote && (
      <div>
        <p>
          Quote:
          { Array.isArray(quote) ? quote[0].quote : quote.quote }
        </p>
        <p>
          Author:
          {Array.isArray(quote) ? quote[0].author : quote.author}
        </p>
      </div>
      )}
    </div>
  );
}

export default Quotes;
