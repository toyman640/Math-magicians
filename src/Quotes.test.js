import React from 'react';
import renderer from 'react-test-renderer';
import fetchMock from 'jest-fetch-mock';
import { act } from 'react-dom/test-utils';
import Quotes from './components/Quotes';

it('renders quote and author from API data', async () => {
  fetchMock.mockResponseOnce(
    JSON.stringify([
      {
        quote: 'Sample quote',
        author: 'Sample author',
      },
    ]),
  );

  let component;

  await act(async () => {
    component = renderer.create(<Quotes />);
  });

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
