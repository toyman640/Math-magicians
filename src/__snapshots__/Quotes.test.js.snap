// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes renders loading state correctly 1`] = `
<p
  className="Quote"
>
  Loading...
</p>
`;

exports[`renders quote and author from API data 1`] = `
<p
  className="Quote"
>
  Loading...
</p>
`;
